 resip/dum/Dialog.cxx | 23 ++++++++++-------------
 1 file changed, 10 insertions(+), 13 deletions(-)

diff --git a/resip/dum/Dialog.cxx b/resip/dum/Dialog.cxx
index 390a05d32..ab8179c8c 100644
--- a/resip/dum/Dialog.cxx
+++ b/resip/dum/Dialog.cxx
@@ -578,17 +578,7 @@ Dialog::dispatch(const SipMessage& msg)
          break;
          case REFER:
          {
-//             if (mInviteSession == 0)
-//             {
-//                InfoLog (<< "Received an in dialog refer in a non-invite dialog: " << request.brief());
-//                SipMessage failure;
-//                makeResponse(failure, request, 603);
-//                mDum.sendResponse(failure);
-//                return;
-//             }
-//             else 
-
-            if  (!request.exists(h_ReferTo))
+            if (!request.exists(h_ReferTo))
             {
                InfoLog (<< "Received refer w/out a Refer-To: " << request.brief());
                SipMessage failure;
@@ -604,10 +594,17 @@ Dialog::dispatch(const SipMessage& msg)
                      (request.exists(h_Requires) &&
                      request.header(h_Requires).find(Token("norefersub"))))
                {
-                  resip_assert(mInviteSession);
+                  if (mInviteSession == 0)
+                  {
+                     InfoLog(<< "Received an in dialog refer with no subcription in a non-invite dialog: " << request.brief());
+                     SipMessage failure;
+                     makeResponse(failure, request, 603);
+                     mDum.sendResponse(failure);
+                     return;
+                  }
                   mInviteSession->referNoSub(msg);
                }
-               else
+               else  // else we need a subscription
                {
                   ServerSubscription* server = findMatchingServerSub(request);
                   ServerSubscriptionHandle serverHandle;
