Index: configure.ac
===================================================================
RCS file: /cvsroot/gaim/gaim/configure.ac,v
retrieving revision 1.248
diff -u -r1.248 configure.ac
--- configure.ac	12 May 2004 03:42:13 -0000	1.248
+++ configure.ac	13 May 2004 17:37:10 -0000
@@ -129,7 +129,7 @@
 fi
 
 if test "x$STATIC_PRPLS" = "xall" ; then
-	STATIC_PRPLS="gg irc jabber msn napster novell oscar silc yahoo zephyr"
+	STATIC_PRPLS="gg irc jabber msn napster novell oscar silc yahoo zephyr simple"
 fi
 if test "x$silcincludes" != "xyes" -o "x$silcclient" != "xyes"; then
 	STATIC_PRPLS=`echo $STATIC_PRPLS | $sedpath 's/silc//'`
@@ -156,6 +156,7 @@
 		trepia)		static_trepia=yes ;;
 		yahoo)		static_yahoo=yes ;;
 		zephyr)		static_zephyr=yes ;;
+		simple)		static_simple=yes ;;
 		*)			echo "Invalid static protocol $i!!" ; exit ;;
 	esac
 done
@@ -172,13 +173,14 @@
 AM_CONDITIONAL(STATIC_TREPIA, test "x$static_trepia" = "xyes")
 AM_CONDITIONAL(STATIC_YAHOO, test "x$static_yahoo" = "xyes")
 AM_CONDITIONAL(STATIC_ZEPHYR, test "x$static_zephyr" = "xyes")
+AM_CONDITIONAL(STATIC_SIMPLE, test "x$static_simple" = "xyes")
 AC_SUBST(STATIC_LINK_LIBS)
 AC_DEFINE_UNQUOTED(STATIC_PROTO_INIT, $extern_init void static_proto_init() { $load_proto },
 	[Loads static protocol plugin module initialization functions.])
 
 AC_ARG_WITH(dynamic_prpls,   [  --with-dynamic-prpls    specify which protocols to build dynamically],[DYNAMIC_PRPLS=`echo $withval | $sedpath 's/,/ /g'`])
 if test "x$DYNAMIC_PRPLS" = "xall" ; then
-	DYNAMIC_PRPLS="gg irc jabber msn napster novell oscar silc yahoo zephyr"
+	DYNAMIC_PRPLS="gg irc jabber msn napster novell oscar silc yahoo zephyr simple"
 fi
 if test "x$silcincludes" != "xyes" -o "x$silcclient" != "xyes"; then
 	DYNAMIC_PRPLS=`echo $DYNAMIC_PRPLS | $sedpath 's/silc//'`
@@ -199,6 +201,7 @@
 		trepia)		dynamic_trepia=yes ;;
 		yahoo)		dynamic_yahoo=yes ;;
 		zephyr)		dynamic_zephyr=yes ;;
+		simple)		dynamic_simple=yes ;;
 		*)			echo "Invalid dynamic protocol $i!!" ; exit ;;
 	esac
 done
@@ -215,6 +218,7 @@
 AM_CONDITIONAL(DYNAMIC_TREPIA, test "x$dynamic_trepia" = "xyes")
 AM_CONDITIONAL(DYNAMIC_YAHOO, test "x$dynamic_yahoo" = "xyes")
 AM_CONDITIONAL(DYNAMIC_ZEPHYR, test "x$dynamic_zephyr" = "xyes")
+AM_CONDITIONAL(DYNAMIC_SIMPLE, test "x$dynamic_simple" = "xyes")
 
 AC_ARG_ENABLE(audio,   [  --disable-audio         compile without libao/libaudiofile for sound playing],,enable_audio=yes)
 AC_ARG_ENABLE(nas,     [  --enable-nas            enable NAS (Network Audio System) support],,enable_nas=no)
@@ -1149,6 +1153,7 @@
 		   src/protocols/trepia/Makefile
 		   src/protocols/yahoo/Makefile
 		   src/protocols/zephyr/Makefile
+		   src/protocols/simple/Makefile
 		   gaim.spec
 		  ])
 
Index: pixmaps/status/default/Makefile.am
===================================================================
RCS file: /cvsroot/gaim/gaim/pixmaps/status/default/Makefile.am,v
retrieving revision 1.20
diff -u -r1.20 Makefile.am
--- pixmaps/status/default/Makefile.am	17 Apr 2004 13:55:27 -0000	1.20
+++ pixmaps/status/default/Makefile.am	13 May 2004 17:37:10 -0000
@@ -30,7 +30,8 @@
 	trepia.png \
 	wireless.png \
 	yahoo.png \
-	zephyr.png
+	zephyr.png \
+	sippy.png
 
 gaimstatuspixdir = $(datadir)/pixmaps/gaim/status/default
 
Index: src/protocols/Makefile.am
===================================================================
RCS file: /cvsroot/gaim/gaim/src/protocols/Makefile.am,v
retrieving revision 1.16
diff -u -r1.16 Makefile.am
--- src/protocols/Makefile.am	2 May 2004 17:57:10 -0000	1.16
+++ src/protocols/Makefile.am	13 May 2004 17:37:10 -0000
@@ -1,3 +1,3 @@
-DIST_SUBDIRS = gg irc jabber msn napster novell oscar rendezvous silc toc trepia yahoo zephyr
+DIST_SUBDIRS = gg irc jabber msn napster novell oscar rendezvous silc toc trepia yahoo zephyr simple
 
 SUBDIRS = $(DYNAMIC_PRPLS) $(STATIC_PRPLS)

Index: Makefile.mingw
===================================================================
RCS file: /cvsroot/gaim/gaim/Makefile.mingw,v
retrieving revision 1.37
diff -u -r1.37 Makefile.mingw
--- Makefile.mingw	17 Apr 2004 13:55:26 -0000	1.37
+++ Makefile.mingw	24 Jun 2004 04:00:50 -0000
@@ -24,6 +24,7 @@
 TREPIA = $(GAIM_PROTOS)/trepia
 GG = $(GAIM_PROTOS)/gg
 NOVELL = $(GAIM_PROTOS)/novell
+SIMPLE = $(GAIM_PROTOS)/simple
 PO = ./po
 
 VERSION := $(shell cat ./VERSION)
@@ -53,6 +54,7 @@
 GG_TYPE = PLUGIN
 NOVELL_TYPE = PLUGIN
 TREPIA_TYPE = PLUGIN
+SIMPLE_TYPE = PLUGIN
 
 all:
 	cp config.h.mingw config.h
@@ -63,6 +65,7 @@
 	$(MAKE) TYPE='$(JABBER_TYPE)' -C $(JABBER) -f Makefile.mingw
 	$(MAKE) TYPE='$(GG_TYPE)' -C $(GG) -f Makefile.mingw
 	$(MAKE) TYPE='$(NOVELL_TYPE)' -C $(NOVELL) -f Makefile.mingw
+	$(MAKE) TYPE='$(SIMPLE_TYPE)' -C $(SIMPLE) -f Makefile.mingw
 	$(MAKE) -C $(GAIM_SRC) -f Makefile.mingw
 	$(MAKE) -C $(GAIM_PLUGINS) -f Makefile.mingw
 
@@ -81,8 +84,11 @@
 	$(MAKE) TYPE='$(JABBER_TYPE)' -C $(JABBER) -f Makefile.mingw install
 	$(MAKE) TYPE='$(GG_TYPE)' -C $(GG) -f Makefile.mingw install
 	$(MAKE) TYPE='$(NOVELL_TYPE)' -C $(NOVELL) -f Makefile.mingw install
+	$(MAKE) TYPE='$(SIMPLE_TYPE)' -C $(SIMPLE) -f Makefile.mingw install
 	cp $(NEEDED_DLLS) $(GAIM_INSTALL_DIR)
 	cp $(SOUNDS) $(GAIM_INSTALL_DIR)/sounds/gaim
+	cp /usr/bin/cygwin1.dll $(GAIM_INSTALL_DIR)
+	cp /usr/local/bin/gag.exe $(GAIM_INSTALL_DIR)
 
 installer:
 	makensis.exe /DGAIM_VERSION="$(VERSION)" /DWITH_GTK gaim-installer.nsi
@@ -105,6 +111,7 @@
 	$(MAKE) -C $(JABBER) -f Makefile.mingw clean
 	$(MAKE) -C $(GG) -f Makefile.mingw clean
 	$(MAKE) -C $(NOVELL) -f Makefile.mingw clean
+	$(MAKE) -C $(SIMPLE) -f Makefile.mingw clean
 	$(MAKE) -C $(GAIM_SRC) -f Makefile.mingw clean
 	$(MAKE) -C $(GAIM_PLUGINS) -f Makefile.mingw clean
 	rm -rf config.h $(GAIM_INSTALL_DIR)

Index: gaim-installer.nsi
===================================================================
RCS file: /cvsroot/gaim/gaim/gaim-installer.nsi,v
retrieving revision 1.66
diff -u -r1.66 gaim-installer.nsi
--- gaim-installer.nsi	30 May 2004 21:54:05 -0000	1.66
+++ gaim-installer.nsi	24 Jun 2004 04:03:59 -0000
@@ -559,6 +559,7 @@
     Delete "$INSTDIR\plugins\liboscar.dll"
     Delete "$INSTDIR\plugins\libtoc.dll"
     Delete "$INSTDIR\plugins\libyahoo.dll"
+    Delete "$INSTDIR\plugins\libsimple.dll"
     Delete "$INSTDIR\plugins\perl.dll"
     Delete "$INSTDIR\plugins\relnot.dll"
     Delete "$INSTDIR\plugins\spellchk.dll"
@@ -594,6 +595,10 @@
     Delete "$INSTDIR\exchndl.dll"
     !endif
 
+    ; Additional files required to support SIMPLE plugin
+    Delete "$INSTDIR\gag.exe"
+    Delete "$INSTDIR\cygwin1.dll"
+
     ;Try to remove Gaim install dir .. if empty
     RMDir "$INSTDIR"
 
