--- c/src/proactor/epoll.c.bak
+++ c/src/proactor/epoll.c
@@ -133,7 +133,7 @@
   pthread_mutexattr_t attr;
 
   pthread_mutexattr_init(&attr);
-  pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_ADAPTIVE_NP);
+  pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_NORMAL);
   if (pthread_mutex_init(pm, &attr)) {
     perror("pthread failure");
     abort();
--- CMakeLists.txt.bak
+++ CMakeLists.txt
@@ -230,7 +230,7 @@
 
 # Start of variables used during install
 set (INCLUDE_INSTALL_DIR include CACHE PATH "Include file directory")
-set (LIB_INSTALL_DIR "lib${LIB_SUFFIX}" CACHE PATH "Library object file directory")
+set (LIB_INSTALL_DIR lib CACHE PATH "Library object file directory")
 set (SYSCONF_INSTALL_DIR etc CACHE PATH "System read only configuration directory")
 set (SHARE_INSTALL_DIR share CACHE PATH "Shared read only data directory")
 set (MAN_INSTALL_DIR share/man CACHE PATH "Manpage directory")
