--- CMakeLists.txt.bak
+++ CMakeLists.txt
@@ -142,7 +142,7 @@
 endif()
 
 set(BINDIR "bin" CACHE PATH "The directory to install binaries into.")
-set(LIBDIR ${SOCI_LIBDIR} CACHE PATH "The directory to install libraries into.")
+set(LIBDIR "lib" CACHE PATH "The directory to install libraries into.")
 set(DATADIR "share" CACHE PATH "The directory to install data files into.")
 set(INCLUDEDIR "include" CACHE PATH "The directory to install includes into.")
 
