Index: GeoIP.c
===================================================================
--- GeoIP.c	(revision 9431)
+++ GeoIP.c	(working copy)
@@ -26,6 +26,32 @@
 #include <unistd.h>
 #include <netdb.h>
 #include <sys/mman.h>
+#else
+#define ssize_t SSIZE_T
+#include <io.h>
+#include <windows.h>
+
+ssize_t pread (int fd, void *buf, size_t count, off_t offset)
+{
+    HANDLE handle = (HANDLE)(intptr_t)_get_osfhandle (fd);
+
+    OVERLAPPED olap = { 0 }; 
+    DWORD written;
+
+    if (handle == INVALID_HANDLE_VALUE)
+        return -1;
+
+    olap.Offset = offset;
+    //olap.OffsetHigh = offset >> 32;
+
+    /* This braindead API will override the file pointer even if we specify
+     * an explicit read offset... So do not expect this to mix well with
+     * regular read() calls. */
+    if (ReadFile (handle, buf, count, &written, &olap))
+        return written;
+    return -1;
+}
+
 #endif /* !defined(_WIN32) */ 
 
 #include <errno.h>
Index: GeoIPCity.c
===================================================================
--- GeoIPCity.c	(revision 9431)
+++ GeoIPCity.c	(working copy)
@@ -38,8 +38,8 @@
 static
 const int       FULL_RECORD_LENGTH = 50;
 
-static const int CITYCONFIDENCE_FIXED_RECORD = 4;
-static const int CITYCONFIDENCEDIST_FIXED_RECORD = 6;
+#define CITYCONFIDENCE_FIXED_RECORD 4
+#define CITYCONFIDENCEDIST_FIXED_RECORD 6
 
 
 static
@@ -55,6 +55,7 @@
   double          latitude = 0, longitude = 0;
   int             metroarea_combo = 0;
   int             bytes_read = 0;
+  int             t = 0;
   if (seek_record == gi->databaseSegments[0])
     return NULL;
 
@@ -93,11 +94,12 @@
       record->city_conf = tmp_fixed_record[2];
       record->postal_conf = tmp_fixed_record[3];
 
+      t = fixed_rec_size - gi->record_length;
+
       record->accuracy_radius =
 	gi->databaseType == GEOIP_CITYCONFIDENCEDIST_EDITION
 	? ((tmp_fixed_record[4] + (tmp_fixed_record[5] << 8)) & 0x3ff) : 0x3ff;
 
-      int             t = fixed_rec_size - gi->record_length;
       
     record_pointer = dseg + tmp_fixed_record[t] +
 	(tmp_fixed_record[t + 1] << 8) + (tmp_fixed_record[t + 2] << 16) ;
@@ -129,7 +131,7 @@
 	gi->databaseType == GEOIP_CITYCONFIDENCEDIST_EDITION
 	? ((record_buf[4] + (record_buf[5] << 8)) & 0x3ff) : 0x3ff;
 
-      int             t = fixed_rec_size - gi->record_length;
+      t = fixed_rec_size - gi->record_length;
 
         record_pointer = dseg + record_buf[t] +
 	(record_buf[t + 1] << 8) + (record_buf[t + 2] << 16) ;
